pkg/cloudprovider/yandex/routes.go

```diff
@@ -77,11 +77,22 @@ func (yc *Cloud) CreateRoute(ctx context.Context, _ string, _ string, route *clo
 	}
 
 	kubeNodeName := string(route.TargetNode)
-	nextHop, err := yc.getInternalIpByNodeName(kubeNodeName)
+	addresses, err := yc.NodeAddresses(ctx, route.TargetNode)
```

Comment:
We switched from getting info about Node IPs from Kubernetes to getting it directly from the cloud API.